src/NuGetForUnity/Editor/Configuration/ConfigurationManager.cs

```diff
@@ -235,16 +236,28 @@ public static Task<List<INugetPackage>> SearchAsync(
         /// <param name="includePrerelease">True to include pre-release packages (alpha, beta, etc).</param>
         /// <param name="targetFrameworks">The specific frameworks to target?.</param>
         /// <param name="versionConstraints">The version constraints?.</param>
+        /// <param name="token">The cancellation token?.</param>
         /// <returns>A list of all updates available.</returns>
         [NotNull]
         [ItemNotNull]
         public static List<INugetPackage> GetUpdates(
             [NotNull] IEnumerable<INugetPackage> packagesToUpdate,
             bool includePrerelease = false,
             string targetFrameworks = "",
-            string versionConstraints = "")
+            string versionConstraints = "",
+            CancellationToken token = default)
         {
-            return ActivePackageSource.GetUpdates(packagesToUpdate, includePrerelease, targetFrameworks, versionConstraints);
+            // Fetching package updates often requires a lot of requests so we increase the limit to speed them up.
+            var oldConnectionLimit = ServicePointManager.DefaultConnectionLimit;
+            try
+            {
+                ServicePointManager.DefaultConnectionLimit = 20;
```

Comment:
This will only affect fetching from nuget source with V2 for V3 we would need to change the setting on the `HttpClient`. Not sure if we should also change it for V3.

Reply:
I tried setting MaxConnectionsPerServer on HttpHandler that we use but I got NotImplementedException. It seems HttpClient in Unity Editor is using MonoHttpHandler which doesn't support that. In my testing this is the only option I found that actually sped up fetching updates over V3.

Reply:
ok could be that Mono even uses `ServicePointManager.DefaultConnectionLimit` internally